### ISSUE TITLE
Problems with strftime and glibc

### ISSUE THREAD

**Issue by andrewyatz** [Linuxbrew/brew#92]
# Please follow the general troubleshooting steps first:
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?
### Bug reports:

I have found an error in compiling a Perl library for working with strftime. This was related to always reporting the timezone and offsets as `+0000` no matter what TZ was exported to. On first observation it seems that glibc/POSIX is not picking up my timezone settings which is in the normal place `/usr/share/zoneinfo/`.

Is there a known issue or can anyone advise how to convince glibc to look at `/usr/share/zoneinfo`? I've looked into reinstalling glibc but that throws an error with `Error: No such file or directory - /nfs/software/linuxbrew/opt/gcc/bin/gcc-5`.

Pretty much at a loss about where/what else to look at so pointers are much appreciated.
### Replicating the error outside of Perl

I was able to replicate this error using the following C code

``` c
#include <time.h>
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[])
{
    char outstr[200];
    time_t t;
    struct tm *tmp;

    t = time(NULL);
    tmp = localtime(&t);
    if (tmp == NULL) {
        perror("localtime");
        exit(EXIT_FAILURE);
    }

     if (strftime(outstr, sizeof(outstr), argv[1], tmp) == 0)
     {
         fprintf(stderr, "strftime returned 0");
         exit(EXIT_FAILURE);
     }

     printf("Result string is \"%s\"\n", outstr);

     exit(EXIT_SUCCESS);
}
```

I compile this using `gcc -o tz tz.c` and it is run specifying `tz 'strftime_string'` e.g. `tz '%Z%z`. That gives the following outputs

```
$ /tmp/tz '%Z%z'
Result string is "BST+0100"
$ TZ='Europe/London' /tmp/tz '%Z%z'
Result string is "Europe+0000"
$ TZ='Europe/Paris' /tmp/tz '%Z%z'
Result string is "Europe+0000"
$ TZ='Australia/Darwin' /tmp/tz '%Z%z'
Result string is "Australia+0000"
```

**Comment by sjackman** [comment]
I can confirm that I see the same unexpected behaviour as you.

**Comment by sjackman** [comment]
This issue reports the same bug: https://github.com/Linuxbrew/legacy-linuxbrew/issues/390

**Comment by sjackman** [comment]
Linux From Scratch explains how to add the time zone data: `6.9.2.2. Adding time zone data`
http://www.linuxfromscratch.org/lfs/view/development/chapter06/glibc.html

**Comment by sjackman** [comment]
This workaround works:

```
$ TZ=PST8PDT ./tz '%Z%z'
Result string is "PDT-0700"
```

**Comment by sjackman** [comment]
I thought this hack may work, ~~but it did not work for me:~~

```
cp -a /usr/share/zoneinfo/ ~/.linuxbrew/opt/glibc/share/
```

_Edit_: This fix does work.

**Comment by andrewyatz** [comment]
Aye I think I tried that and was unsuccessful. Perhaps I'm bouncing into
issues with trying to reuse an existing zoneinfo build that shipped with my
Linux (it's RHEL7.2). Will let you know how I get on.

**Comment by sjackman** [comment]
If you follow the instructions from Linux From Scratch that's probably your best bet to get it working.  Or the workaround `TZ=PST8PDT` is pretty easy.

**Comment by andrewyatz** [comment]
I'll go and try the LfS instructions since it's causing issues downstream
with a Perl test suite and I know this means the more descriptive timezone
locations won't work. Will update the issue as and when I get this solved.
Cheers for the pointers.

**Comment by andrewyatz** [comment]
Hi. Posting an update on this (trying to make sure I don't do a DenverCoder9). I did some more sniffing around with the compiled `libc.so.6` and found where it expects zoneinfo to live.

``` bash
$ strings $(brew --prefix glibc)/lib/libc.so.6 | grep 'zoneinfo'
/nfs/software/linuxbrew/Cellar/glibc/2.19/share/zoneinfo
```

Running `ln -s /usr/share/zoneinfo $(brew --prefix glibc)/share` suddenly made glibc behave

``` bash
$ TZ='Australia/Darwin' /tmp/tz '%Z%z'
Result string is "ACST+0930"
```

I'll create a PR to get this working for glibc in a similar way you solved it for Linuxbrew/legacy-linuxbrew#390. However as I said before I've been struggling to get glibc to recompile on my linuxbrew installation so this will be experimental (the problem with it not finding gcc-5).

**Comment by sjackman** [comment]
I don't know what I did wrong when I tested, `cp -a /usr/share/zoneinfo/ ~/.linuxbrew/opt/glibc/share/` above, but it does work. Your symlink also works. A PR similar to c569aa8d12cb6566b70968fb5f82b1c272edecf6 would be great!
